src/Yaapii.Atoms/Number/NumberOf.cs

```diff
@@ -93,53 +93,68 @@ public NumberOf(string str, IFormatProvider provider) : this(str, new ScalarOf<I
         /// </summary>
         /// <param name="str">The string</param>
         /// <param name="provider">a number format provider</param>
-        public NumberOf(string str, IScalar<IFormatProvider> provider) : this(
+        public NumberOf(string str, IScalar<IFormatProvider> provider) : this(new TextOf(str), provider)
+        { }
+
+        /// <summary>
+        /// A <see cref="int"/> as a <see cref="INumber"/>
+        /// </summary>
+        /// <param name="text">The text</param>
+        public NumberOf(IText text) : this(text, new ScalarOf<IFormatProvider>(() => CultureInfo.InvariantCulture))
```

Comment:
CultureInfo.InvariantCulture is not code execution. So you can get rid of Scalar.